From 586cca384be6710b03e14bcbeb7588c1772604ec Mon Sep 17 00:00:00 2001
From: iglocska <andras.iklody@gmail.com>
Date: Thu, 28 Mar 2019 12:05:21 +0100
Subject: [PATCH] fix: [security] Fix to a reflected XSS in the default layout
 template

- as reported by Tuscany Internet eXchange | Misp Team | TIX CyberSecurity
---
 app/View/Layouts/default.ctp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/app/View/Layouts/default.ctp b/app/View/Layouts/default.ctp
index 9b720d2958..93139d9c03 100644
--- a/app/View/Layouts/default.ctp
+++ b/app/View/Layouts/default.ctp
@@ -101,9 +101,9 @@
         var baseurl = '<?php echo $baseurl; ?>';
         var here = '<?php
                 if (substr($this->params['action'], 0, 6) === 'admin_') {
-                    echo $baseurl . '/admin/' . $this->params['controller'] . '/' . substr($this->params['action'], 6);
+                    echo $baseurl . '/admin/' . h($this->params['controller']) . '/' . h(substr($this->params['action'], 6));
                 } else {
-                    echo $baseurl . '/' . $this->params['controller'] . '/' . $this->params['action'];
+                    echo $baseurl . '/' . h($this->params['controller']) . '/' . h($this->params['action']);
                 }
             ?>';
         $(document).ready(function(){
